pkg/networkdevice/profile/profiledefinition/metadata.go

```diff
@@ -8,8 +8,49 @@ package profiledefinition
 // MetadataDeviceResource is the device resource name
 const MetadataDeviceResource = "device"
 
-// MetadataConfig holds configs per resource type
-type MetadataConfig map[string]MetadataResourceConfig
+// MetadataConfig holds metadata config per resource type
+type MetadataConfig struct {
+	Device    DeviceMetadata    `yaml:"device,omitempty" json:"device,omitempty"`
+	Interface InterfaceMetadata `yaml:"interface,omitempty" json:"interface,omitempty"`
+}
```

Comment:
There are several other metadata sections, all defined in [config_metadata.go](https://github.com/DataDog/datadog-agent/blob/main/pkg/collector/corechecks/snmp/internal/checkconfig/config_metadata.go) - although the Device section is only one we let users edit, we automatically populate `interfaces` and `ip_addresses` on all metadata and `lldp_remote`, `lldp_remote_management`, `lldp_local`, and `cdp_remote` if topology is enabled.

There's a list of all metadata fields I could find mentions of [in confluence](https://datadoghq.atlassian.net/l/cp/JNp1UYT0) as well.

I assume the reason these aren't present in `validMetadataResources` is because we never set them via actual profile files - they're fixed fields that are added to the metadata when topology is enabled because the code in `report_devcie_metadata.go` needs them to be present, and they're all defined by various RFCs so there shouldn't be cases where anyone needs to change how they're fetched.

Reply:
Possibly it would be a good idea to have two separate "metadata" structures - one that *just* has the device fields, since those are the ones configurable by users, and one that is a generic list of symbols to fetch and that isn't encoded in yaml or json? That way the profile structure that gets edited, saved to disk, etc. just has the device fields, but internally we can still tack on the list of topology fields when necessary? E.g. we don't need entries like
```go
	"lldp_remote_management": {
		Fields: map[string]profiledefinition.MetadataField{
			"interface_id_type": {
				Symbol: profiledefinition.SymbolConfig{
					OID:  "1.0.8802.1.1.2.1.4.2.1.3",
					Name: "lldpRemManAddrIfSubtype",
				},
			},
		},
	},
```
when we could just have an entry in a list 
```
SymbolConfig{
	OID:  "1.0.8802.1.1.2.1.4.2.1.3",
	Name:  "lldp_remote_management.interface_id_type",
}
```